### PR TITLE
 [lab-customer-analysis-final-round]izelyekrek

### PR DESCRIPTION
**Pull request by izelyekrek**

### CODE REVIEW

**Comment by fromeropablo** [comment]
# lab-customer-analysis-round-6 + final-round

Hi Izel! As you did the last two labs during last weekend, I'll give you the correction from both labs here.

That's the last step to understand better Machine Learning basics at the beginning of the course. Summed to the work done in the previous labs, here you had to familiarized with categorical transformations such as **get_dummies** or **OneHotEncoder**. 

Also, you had to calculate some important metrics from Linear Regressions and understand what are explaining. 

## 1. Previous work

**Dealing with the data.** Good work here checking for missing values and standardizing your columns names. As I told to you in the last lab, Pandas provides us with a helpful function called `.duplicated` to check for repeated rows before removing them. It's your decision to keep those rows in your dataset or not.

**Cleaning/Wrangling/EDA** Take in count here that you have to do all the cleaning that you have been doing in the previous labs: check correlations, duplicates, missing values, data type...

You did it almost everything, really good work!! You forgot to do a correlation matrix to check the relation between features and the target. Remember that it's better to visualize create a correlation matrix using `.heatmap` from seaborn. 

## 2. Processing data

Great work in this part in both labs! In the first, be careful to create and run a Linear Regression without normalize your features and your target and enconde your categorical variables. It will give you, normally, worse results than not following these steps before.

Take in count that you don't need to use `pickle` yet. It's a library that allows you to save your progress in a file. It's not necessary to normalize your variables.

Well done using *OneHotEncoder* from `sklearn` module to create numerical variables from categorical ones! Remember that, depending on the categorical variable, you can use *get_dummies* aswell. 

Be careful here with the arrays. You just created the arrays properly but you had to finish the task creating a DataFrame from those arrays. Your code, optimized, would be something like this:
```python 
encoder = pd.DataFrame(OneHotEncoder(handle_unknown='error', drop='first').fit_transform(X_cat_new).toarray())
```
Doing that, it seems that you should concat both DataFrames, numerical and encoder, using the Pandas function `pd.concat`. It's better to visualize your data than `np.concatenate` which only concatenate arrays. 

Great work using a function to remove the outliers, congrats! The last questions are perfect aswell, you had to create a Linear Regression Model after do all the cleaning in your data and create a predicted target. 

Then, the metrics helps us to understand if our model is well explained by the features or not. 

Your report is really good, but just a few things:

- You can change the format of that cell from *Code* to *Markdown* and remove all the hashes at the beginning of each sentence.

- For the future, you have to give a better explanation on why you are removing some variables. You cannot say that you think they're not important in the model, we are Data Analysts so we need to explain it in a quantitative or qualitative way.
